tests/test_cookie.py

```diff
@@ -0,0 +1,193 @@
+import base64
+import unittest
+from flask import Flask
+from flask_httpauth import HTTPCookieAuth
+
+
+class HTTPAuthTestCase(unittest.TestCase):
+    def setUp(self):
+        app = Flask(__name__)
+        app.config['SECRET_KEY'] = 'my secret'
+
+        cookie_auth = HTTPCookieAuth('MyToken')
+        cookie_auth2 = HTTPCookieAuth('Token', realm='foo')
+        cookie_auth3 = HTTPCookieAuth(scheme='ApiKey', cookie_name='X-API-Key')
+        cookie_default = HTTPCookieAuth()
+
+        @cookie_auth.verify_cookie
+        def verify_cookie(token):
+            if token == 'this-is-the-token!':
+                return 'user'
+
+        @cookie_auth3.verify_cookie
+        def verify_cookie3(token):
+            if token == 'this-is-the-token!':
+                return 'user'
+
+        @cookie_default.verify_cookie
+        def verify_cookie_default(token):
+            if token == 'this-is-the-token!':
+                return 'user'
+
+        @cookie_auth.error_handler
+        def error_handler():
+            return 'error', 401, {'WWW-Authenticate': 'MyToken realm="Foo"'}
```

Comment:
This is really not something I can agree with. Here you are mixing your own custom authentication implementation based on cookies with parts of the HTTP Authentication standard, which uses the `Authorization` header as vehicle for the client to send credentials. There is no good reason to return the `WWW-Authenticate` header that I can see and I have never seen any implementation that does it.